### PR TITLE
Migrate define(OS_*) to using BUILDFLAG(IS_*) instead

### PR DESCRIPTION
**Pull request by mariospr**
See the following PSA in the chromium-dev mailing list for more
information and links to the original discussion, the design
document and the metabug:

https://groups.google.com/a/chromium.org/g/chromium-dev/c/EIuQCy9sVfw/m/Net3FPz6AwAJ

Resolves https://github.com/brave/brave-browser/issues/21608

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

N/A

### CODE REVIEW

File: browser/brave_rewards/rewards_tab_helper.cc [hunk 1]
```diff
@@ -9,7 +9,7 @@
 #include "brave/components/brave_rewards/browser/rewards_service.h"
 #include "brave/components/ipfs/buildflags/buildflags.h"
 #include "chrome/browser/profiles/profile.h"
-#if !defined(OS_ANDROID)
+#if !BUILDFLAG(IS_ANDROID)
```
Comment (emerick) [inline browser/brave_rewards/rewards_tab_helper.cc:12]: Do we need to include "build/build_config.h" or something similar when referencing this?

**Review by AlexeyBarabash — APPROVED** [review]
++ for sync changes

**Review by simonhong — APPROVED** [review]
++ but have same concerns that @emerick mentioned - including `build/build_config.h`

File: browser/brave_rewards/rewards_tab_helper.cc (hunk 1 above)
Comment (mariospr) [reply browser/brave_rewards/rewards_tab_helper.cc:12]: Yes, we should. I added it already on the places where it was not explicitly included but it seems I missed 
"quite a few" more, thanks for the catch. Fixed in https://github.com/brave/brave-core/pull/12655/commits/7d7f5e7554abbad3f8d1baaff7d991cbd520bf51

File: components/brave_rewards/browser/rewards_notification_service_impl.cc [hunk 2]
```diff
@@ -351,7 +352,7 @@ void RewardsNotificationServiceImpl::OnFetchPromotions(
     args.push_back(base::NumberToString(item->claimable_until * 1000));
 
     bool only_once = true;
-  #if defined(OS_ANDROID)
+#if BUILDFLAG(IS_ANDROID)
     only_once = false;
   #endif
```
Comment (emerick) [inline components/brave_rewards/browser/rewards_notification_service_impl.cc:357]: Nit: Can we fix up the spacing on the closing #endif as well?

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 3]
```diff
@@ -256,9 +256,9 @@ ledger::type::InlineTipsPlatforms ConvertInlineTipStringToPlatform(
 
 bool ProcessPublisher(const GURL& url) {
   // we should always process publisher on desktop
-  #if !defined(OS_ANDROID)
-    return true;
-  #endif
+#if !BUILDFLAG(IS_ANDROID)
```
Comment (emerick) [inline components/brave_rewards/browser/rewards_service_impl.cc:259]: I think we need the include in this file?

File: components/brave_rewards/browser/service_sandbox_type.h [hunk 4]
```diff
@@ -17,11 +17,11 @@ class BatLedgerService;
 template <>
 inline sandbox::mojom::Sandbox
 content::GetServiceSandboxType<bat_ledger::mojom::BatLedgerService>() {
-#if !defined(OS_ANDROID)
+#if !BUILDFLAG(IS_ANDROID)
```
Comment (emerick) [inline components/brave_rewards/browser/service_sandbox_type.h:20]: I think we need the include in this file?

File: components/brave_rewards/browser/rewards_notification_service_impl.cc (hunk 2 above)
Comment (mariospr) [reply components/brave_rewards/browser/rewards_notification_service_impl.cc:357]: oops! I thought `git cl format` would fix that but it didn't. Fixed in https://github.com/brave/brave-core/commit/23457292cae92ce151138c85b8c80552362c527f

File: components/brave_rewards/browser/service_sandbox_type.h (hunk 4 above)
Comment (mariospr) [reply components/brave_rewards/browser/service_sandbox_type.h:20]: Somehow I missed this one too :facepalm:. Added in https://github.com/brave/brave-core/commit/ffb4a13308e6ebe0c8af573edab9f32f69406a71

File: components/brave_rewards/browser/rewards_service_impl.cc (hunk 3 above)
Comment (emerick) [reply components/brave_rewards/browser/rewards_service_impl.cc:259]: Yup, perfect.

**Review by emerick — APPROVED** [review]
LGTM

**Review by iefremov — APPROVED** [review]

**Review by yrliou — APPROVED** [review]
LGTM

**Review by bsclifton — APPROVED** [review]
Didn't check EVERY change - but spot checked a bunch and this looks great! Thanks for getting us updated, @mariospr 😄

**Pull request merged**